### PR TITLE
Fix for missing Fulltext Image in Newsfeeds

### PR DESCRIPTION
**Pull request by AndySDH**
### Summary of Changes

Originally Joomla RSS Newsfeeds were completely missing the image_intro and image_fulltext. This was partially fixed with this PR: https://github.com/joomla/joomla-cms/pull/11402

But that only introduced the image_intro, and forgot the image_fulltext.

So an article that had an image_fulltext would not use that image in the feed.

This PR simply adds the image_fulltext to the feed if the image_intro is empty.

### CODE REVIEW

**Comment by SharkyKZ** [comment]
That was intentional. Fulltext image is not used in category views.

**Comment by AndySDH** [comment]
> That was intentional. Fulltext image is not used in category views.

Right, but that shouldn't matter. This is a newsfeed of the **articles contained in the category**. So I believe the main image representing the article should show in a feed list of the articles.

**Comment by simbus82** [comment]
I have had this problem too, we work with websites that are in Google News.
It is very important to have a really good feed.
https://support.google.com/news/publisher-center/answer/9545245

File: components/com_content/views/category/view.feed.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ protected function reconcileNames($item)
 		$params            = $app->getParams();
 		$item->description = '';
 		$obj = json_decode($item->images);
-		$introImage = isset($obj->{'image_intro'}) ? $obj->{'image_intro'} : '';
+		$introImage = ($obj->{'image_intro'} != '') ? $obj->{'image_intro'} : ( ($obj->{'image_fulltext'} != '') ? $obj->{'image_fulltext'} : '');
```
Comment (Quy) [inline components/com_content/views/category/view.feed.php]:
```suggestion
		$introImage = ($obj->{'image_intro'} != '') ? $obj->{'image_intro'} : (($obj->{'image_fulltext'} != '') ? $obj->{'image_fulltext'} : '');
```

File: components/com_content/views/category/view.feed.php [hunk 2]
```diff
@@ -39,7 +39,7 @@ protected function reconcileNames($item)
...
 
 		if (isset($introImage) && ($introImage != ''))
```
Comment (Quy) [inline components/com_content/views/category/view.feed.php]:
```suggestion
		if ($introImage != '')
```

File: components/com_content/views/featured/view.feed.php [hunk 3]
```diff
@@ -52,7 +52,7 @@ public function display($tpl = null)
 
 			$description = '';
 			$obj = json_decode($row->images);
-			$introImage = isset($obj->{'image_intro'}) ? $obj->{'image_intro'} : '';
+			$introImage = ($obj->{'image_intro'} != '') ? $obj->{'image_intro'} : ( ($obj->{'image_fulltext'} != '') ? $obj->{'image_fulltext'} : '');
```
Comment (Quy) [inline components/com_content/views/featured/view.feed.php]:
```suggestion
			$introImage = ($obj->{'image_intro'} != '') ? $obj->{'image_intro'} : (($obj->{'image_fulltext'} != '') ? $obj->{'image_fulltext'} : '');
```

File: components/com_content/views/featured/view.feed.php [hunk 4]
```diff
@@ -52,7 +52,7 @@ public function display($tpl = null)
...
 
 			if (isset($introImage) && ($introImage != ''))
```
Comment (Quy) [inline components/com_content/views/featured/view.feed.php]:
```suggestion
			if ($introImage != '')
```

**Comment by marcosliell** [comment]
I need the image path in a separate tag. and not along with the introduction text.

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on a356791b05003cdd7dfba7ffc5ef41ce4d3f55ae<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28195">issues.joomla.org/tracker/joomla-cms/28195</a>.</sub>

**Comment by AndySDH** [comment]
Hello, we need one more successful test for this. Or anything else I need to do to get this merged?

**Comment by gostn** [comment]
I have tested this item :white_check_mark: successfully on a356791b05003cdd7dfba7ffc5ef41ce4d3f55ae<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28195">issues.joomla.org/tracker/joomla-cms/28195</a>.</sub>

**Comment by AndySDH** [comment]
Thanks guys, asking for this to be merged after the successful tests have been made

**Comment by alikon** [comment]
humm, 

drone still failing, but most probably is not related to your pr

File: components/com_content/views/category/view.feed.php [hunk 5]
```diff
@@ -39,9 +39,9 @@ protected function reconcileNames($item)
 		$params            = $app->getParams();
 		$item->description = '';
 		$obj = json_decode($item->images);
-		$introImage = isset($obj->{'image_intro'}) ? $obj->{'image_intro'} : '';
+		$introImage = ($obj->{'image_intro'} != '') ? $obj->{'image_intro'} : (($obj->{'image_fulltext'} != '') ? $obj->{'image_fulltext'} : '');
```
Comment (SharkyKZ) [inline components/com_content/views/category/view.feed.php]: Restore `isset()` check.

File: components/com_content/views/featured/view.feed.php [hunk 6]
```diff
@@ -52,9 +52,9 @@ public function display($tpl = null)
 
 			$description = '';
 			$obj = json_decode($row->images);
-			$introImage = isset($obj->{'image_intro'}) ? $obj->{'image_intro'} : '';
+			$introImage = ($obj->{'image_intro'} != '') ? $obj->{'image_intro'} : (($obj->{'image_fulltext'} != '') ? $obj->{'image_fulltext'} : '');
```
Comment (SharkyKZ) [inline components/com_content/views/featured/view.feed.php]: Restore `isset()` check.

**Comment by AndySDH** [comment]
@SharkyKZ No, the isset check makes no sense. I purposely removed it. The variable already has a default value so there is no way it can ever not be set.

**Comment by SharkyKZ** [comment]
Not for the variable, for the object property.

**Comment by AndySDH** [comment]
It's set as if not empty. if you want to make the change yourself, go ahead. And then merge it.

I'm not gonna make further changes for an insignificant thing, just to wait for more tests, just for you to report another insignificant thing in 6 months from now and repeat the loop again.

**Comment by SharkyKZ** [comment]
I'm not going to make any changes because 1) I don't have such privilege 2) this PR is wrong as already explained in the first comment.

But if maintainers accept this change, it at least should not be producing PHP notices:

>PHP Notice:  Trying to get property of non-object in C:\wamp\www\joomla-cms-3\components\com_content\views\category\view.feed.php on line 42

**Comment by SharkyKZ** [comment]
I'm not going to make any changes because 1) I don't have such privilege 2) this PR is wrong as already explained in the first comment.

But if maintainers accept this change, it at least should not be producing PHP notices:

>PHP Notice:  Trying to get property of non-object in C:\wamp\www\joomla-cms-3\components\com_content\views\category\view.feed.php on line 42

**Comment by AndySDH** [comment]
The PR is not wrong. f you google around you find many people complaining about this same issue.

Anyway, I'll make the changes, I'll also make the code a little bit more nice to read.

**Comment by AndySDH** [comment]
Done, please test @SharkyKZ, thanks.

**Comment by AndySDH** [comment]
Not sure why checks are failing, anyone can help?

**Comment by HLeithner** [comment]
please merge the current staging branch into this pr branch

**Comment by AndySDH** [comment]
@HLeithner can you give me instructions on how to do it? No idea how that's done =/ Thanks.

**Comment by AndySDH** [comment]
Soo... Can someone with the knowledge to do this merge this into the latest branch? Thanks!

**Comment by AndySDH** [comment]
Hello, any update on this?

**Comment by zero-24** [comment]
I'm sorry that it took that long to get a reply here but i will not accept this PR as it is into 3.10. if anything we should do a new PR against 4.2-dev.

**Comment by AndySDH** [comment]
See PR https://github.com/joomla/joomla-cms/pull/38058 for 4.2 re-do

**Pull request closed without merging**